polkadot/runtime/rococo/src/governance/fellowship.rs

```diff
@@ -32,12 +37,13 @@ pub struct TracksInfo;
 impl pallet_referenda::TracksInfo<Balance, BlockNumber> for TracksInfo {
 	type Id = u16;
 	type RuntimeOrigin = <RuntimeOrigin as frame_support::traits::OriginTrait>::PalletsOrigin;
-	fn tracks() -> &'static [(Self::Id, pallet_referenda::TrackInfo<Balance, BlockNumber>)] {
-		static DATA: [(u16, pallet_referenda::TrackInfo<Balance, BlockNumber>); 10] = [
-			(
-				0u16,
-				pallet_referenda::TrackInfo {
-					name: "candidates",
+
+	fn tracks() -> impl Iterator<Item = Cow<'static, Track<Self::Id, Balance, BlockNumber>>> {
```

Comment:
Could it be `RandomAccessIterator` by any chance? Dont know if that is possible, but could be better in cases where a single one needs to be accessed.

Reply:
To access a particular track there is the `TracksInfo::info(id)` method, it's implemented by default iterating over the tracks to `.find` the right one but it can be overwritten. In our ["dynamic tracks" implementation](https://github.com/virto-network/polkadot-sdk/blob/release-virto-v1.5.0/substrate/frame/referenda-tracks/src/impl_tracks_info.rs#L19) for example we replace it with an access to the particular track in the storage.

Reply:
Okay 👍  so more performant implementations could make use of that still.